core/sql/optimizer/SimpleScanOptimizer.cpp

```diff
@@ -218,25 +218,14 @@ SimpleFileScanOptimizer::isLogicalSubPartitioned() const
 SearchKey *
 SimpleFileScanOptimizer::constructSearchKey()
 {
+  // We do not need to incude the flatten version of RANGE SPEC predicates 
+  // to the exePreds because SearchKey is capable of handling such predicates.
```

Comment:
I remember in the MDAM document there are examples where a predicate had to be in mdam disjunct as well as in exe pred to give correct results. This was because mdam disjunct could only guarantee minimal set of rows were returned (in such cases) instead of something that exactly matched the predicate. I can look for the example if it will help.